### PR TITLE
feat: integrate a helm sdk that use to install gateway helm chart

### PR DESCRIPTION
**Pull request by RolandMa1986**
### What type of PR is this?
/kind feature

### What this PR does / why we need it:
As feature #3055 mentioned, we need a flexible way to deploy the kubesphere gateway. One of the solutions is using the Operator model to deploy the related gateway resources. 
In this PR:
1. Helm Operator SDK has been integrated into the ks-controller-manager, we can easily convert any helm chart into an Operator. 
2. An `watches.yaml` config file can be configurated in the gateway options,  which use to monitor the creation of Gateway resource and deployment gateway helm chart.
```
- group: gateway.kubesphere.io
  version: v1alpha1
  kind: Gateway
  chart: config/gateway
```
3. Added a gateway helm that use to render the Nginx Ingress Controller Helm chart.
3. Upgraded Helm package to helm 3.6.1

### Which issue(s) this PR fixes:
Fixes ##3055

### Special notes for reviewers:
```
```

### Does this PR introduced a user-facing change?
```release-note
None
```

### Additional documentation, usage docs, etc.:
```docs

```

### CODE REVIEW

**Comment by RolandMa1986** [comment]
@zryfish Please take a look at this PR.

File: pkg/apiserver/config/config_test.go [hunk 1]
```diff
@@ -179,6 +180,10 @@ func newTestConfig() (*Config, error) {
 		MeteringOptions: &metering.Options{
 			RetentionDay: "7d",
 		},
+		GatewayOptions: &gateway.Options{
+			WatchesPath: "/etc/kubesphere/watches.yaml",
```
Comment (zryfish) [inline pkg/apiserver/config/config_test.go:184]: What does the `watches.yaml` look like, and why do we need it?

File: pkg/apiserver/config/config.go [hunk 2]
```diff
@@ -104,6 +105,7 @@ type Config struct {
 	NotificationOptions   *notification.Options                      `json:"notification,omitempty" yaml:"notification,omitempty" mapstructure:"notification"`
 	KubeEdgeOptions       *kubeedge.Options                          `json:"kubeedge,omitempty" yaml:"kubeedge,omitempty" mapstructure:"kubeedge"`
 	MeteringOptions       *metering.Options                          `json:"metering,omitempty" yaml:"metering,omitempty" mapstructure:"metering"`
+	GatewayOptions        *gateway.Options                           `json:"gateway,omitempty" yaml:"router,omitempty" mapstructure:"router"`
```
Comment (zryfish) [inline pkg/apiserver/config/config.go:108]: Should we use `gateway` for the YAML as well? I am a little worried about the difference here

File: config/gateway/values.yaml [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+service:
+  ## annotations for Services, used for config Cloud LoadBalancer
+  annotations: {}
+  type: LoadBalancer
+
+## for nginx controller, same with controller
+deployment:
+  annotations: {}
+  replicas: 1
```
Comment (zryfish) [inline config/gateway/values.yaml:25]: Should add newline at the end of the file.

File: pkg/apiserver/config/config.go (hunk 2 above)
Comment (RolandMa1986) [reply pkg/apiserver/config/config.go]: Yes, it should be `gateway`. fixed.

File: pkg/apiserver/config/config_test.go (hunk 1 above)
Comment (RolandMa1986) [reply pkg/apiserver/config/config_test.go:184]: Please check the comments in the PR (2nd item). Generally, it's a mapping relation between CRD and the Helm Chart. The operator will monitor the CRD configured in the file, when CR is created and a corresponding Helm Release will be created too.
Currently, the Helm Operator SDK only supports file path-based parameters. We may replace it with CRD in the feature.

File: config/gateway/values.yaml (hunk 3 above)
Comment (RolandMa1986) [reply config/gateway/values.yaml]: fixed.

**Pull request merged**